olp-cpp-sdk-core/src/cache/DefaultCacheImpl.cpp

```diff
@@ -303,6 +303,21 @@ bool DefaultCacheImpl::RemoveKeysWithPrefix(const std::string& key) {
   return true;
 }
 
+bool DefaultCacheImpl::Exist(const std::string& key) {
+  // if lru exist check key only there
+  if (mutable_cache_lru_) {
+    auto iterator = mutable_cache_lru_->FindNoPromote(key);
+    return (iterator != mutable_cache_lru_->end());
+  }
+  if (mutable_cache_ && mutable_cache_->Exist(key)) {
+    return true;
+  }
+  if (protected_cache_ && protected_cache_->Exist(key)) {
+    return true;
+  }
+  return false;
```

Comment:
I believe, need to check memory cache too.
Also no need to check mutable cache if key not found in the lru.